flink-runtime/src/main/java/org/apache/flink/runtime/highavailability/JobResultEntry.java

```diff
@@ -0,0 +1,81 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.runtime.highavailability;
+
+import org.apache.flink.runtime.jobmaster.JobResult;
+import org.apache.flink.runtime.rest.messages.json.JobResultDeserializer;
+import org.apache.flink.runtime.rest.messages.json.JobResultSerializer;
+import org.apache.flink.util.Preconditions;
+
+import org.apache.flink.shaded.jackson2.com.fasterxml.jackson.annotation.JsonCreator;
+import org.apache.flink.shaded.jackson2.com.fasterxml.jackson.annotation.JsonIgnoreProperties;
+import org.apache.flink.shaded.jackson2.com.fasterxml.jackson.annotation.JsonProperty;
+import org.apache.flink.shaded.jackson2.com.fasterxml.jackson.databind.annotation.JsonDeserialize;
+import org.apache.flink.shaded.jackson2.com.fasterxml.jackson.databind.annotation.JsonSerialize;
+
+/**
+ * {@code JobResultEntry} is the entity managed by the {@link JobResultStore}. It collects
+ * information about a globally terminated job (e.g. {@link JobResult}).
+ */
+public class JobResultEntry implements WithJobResult {
+
+    private final JobResult jobResult;
+
+    public JobResultEntry(JobResult jobResult) {
+        this.jobResult = Preconditions.checkNotNull(jobResult);
+    }
+
+    public JobResult getJobResult() {
+        return jobResult;
+    }
+
+    @JsonIgnoreProperties(
+            value = {JsonJobResultEntry.FIELD_NAME_VERSION},
+            allowGetters = true)
+    /**
+     * Wrapper class around {@link JobResultEntry} to allow for serialization of a schema version,
+     * so that future schema changes can be handled in a backwards compatible manner.
+     */
+    public static class JsonJobResultEntry {
```

Comment:
```suggestion
    public static class JsonJobResultEntry extends JobResultEntry {
```
Additionally, you can extract that class into its own file or even make it a private internal class of the `FileSystemJobResultStore`.

Reply:
FYI: What you did was creating an inner class (i.e. a class within a class), not a subclass (a class inheriting from a parent class through `extends`)